OutOfSchool/OutOfSchool.EmailSender/SendGridAccessibilityService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+
+using Quartz;
+
+namespace OutOfSchool.EmailSender;
+
+public interface ISendGridAccessibilityService 
+{
+    bool IsSendGridAccessible { get; set; }
+}
+public class SendGridAccessibilityService : ISendGridAccessibilityService
+{
+    public bool IsSendGridAccessible { get; set; }
+}
+
+public class EmailSenderJobListener : IJobListener
+{
+    private readonly ISendGridAccessibilityService sendGridAccessibilityService;
+
+    public string Name => "Email Sender Job Listener";
+
+    public EmailSenderJobListener(ISendGridAccessibilityService sendGridAccessibilityService)
+    {
+        this.sendGridAccessibilityService = sendGridAccessibilityService;
+    }
+
+    public Task JobExecutionVetoed(IJobExecutionContext context, CancellationToken cancellationToken = default)
+    {
+        return Task.CompletedTask;
+    }
+
+    public Task JobToBeExecuted(IJobExecutionContext context, CancellationToken cancellationToken = default)
+    {
+        return Task.CompletedTask;
+    }
+
+    public Task JobWasExecuted(IJobExecutionContext context, JobExecutionException jobException, CancellationToken cancellationToken = default)
+    {
+        if(jobException != null) 
```

Comment:
(Exception handling) As mentioned in other comment, this condition will never occur. You have to throw `JobExecutionException` manually